commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java

```diff
@@ -34,7 +43,9 @@
 
   @Option(
       names = {"--azure-client-id"},
-      description = "The ID used to authenticate with Azure key vault",
+      description =
+          "The ID used by client secret or user-assigned managed identity authentication mode to access Azure key vault. "
+              + "Optional for system-assigned managed identity.",
```

Comment:
clientId is required when mode is CLIENT_SECRET. If can be left empty (or assign empty value, via config file for instance) for MANAGED_IDENTITY (System-Generated). It can be set to a User-Assigned Managed Identity (hence Optional for Managed Identity). Feel free to suggest alternate wordings.